### DISCUSSION TITLE
z-index of elements

### DISCUSSION THREAD

**Discussion by moving-bits** [mapsforge/vtm#969] [Q&A] [answered]
A quick question, just to be on the safe side: 

z-index of map elements (markers, paths, ...) is not determined by a specific parameter to the element, but by layer order, with elements on a layer with higher index being more visible?

So for example the base tile layer should get a very small index (0 or 1?), the layer for general markers a bit higher (10), and the most important data an even higher index (eg. 20)?

Layers can be added in random order, even when given a specific index number, and index numbers may have gaps?

**Comment by devemux86** [comment, marked as answer]
VTM layers (map, overlays, etc.) work like in Mapsforge, there is list of layers, i.e. z-order.

However, markers in VTM:
- Markers are stored in `ItemizedLayer` classes
- VTM is 3D: markers are sorted on y-axis in real time (the closest one appears in front)
(it can be disabled with `Parameters.MARKER_SORT` and manage the markers list manually)

**Reply by moving-bits** [reply]
I have quite a few different items on the map: A symbol representing the current position would be top-most (`ItemizedLayer`), different paths for tracks or routes go next (a couple of `PathLayer`), then markers for geocaches (`ItemizedLayer`), then those for waypoints (`ItemizedLayer`), and some more less-important stuff below that (`VectorLayer`, `PathLayer` etc.). Ordering within each group of markers can be left to automatic.

I have done some further testing meanwhile and found out, that layer indexes have to have consecutive numbers, no gaps allowed, so I need to make sure, that I don't require a variable number of layers.

Which leads me to a follow-up question (as my current approach conflicts with not having a variable number of layers): 

I have a couple of paths to draw (eg.: each path represents an accumulated track history for a specific month). 
- Each path can consist of several unrelated segments (eg.: track history for a single day), 
- each segment consisting of several points (eg.: the places you have visited on that specific day) - this should be an actual path with connected elements.

I'm using a `PathLayer` to display this, but I have not been able to show the segments unconnected, so I'm currently putting each segment into its own layer. Surely there must be a better way to do this.

**Comment by devemux86** [comment]
_For new unrelated questions please create a new discussion._

GPX tracks with multiple track segments: need a path layer per track segment (Mapsforge + VTM).
Map libraries do not care if 2 lines are "related", if not connected then they are separate line layers.

**Reply by moving-bits** [reply]
_(Marked the answer to the original question as answer, and continued my follow-up question in a separate thread: https://github.com/mapsforge/vtm/discussions/970)_
